In [1]:
import pandas as pd
import numpy as np
df=pd.DataFrame()
print(df)

Empty DataFrame
Columns: []
Index: []


In [2]:
my_data={
    'Names':["peter","Mount","Abraham"],
    'Age':[20,18,22]}
dt=pd.DataFrame(my_data)
print(dt)

     Names  Age
0    peter   20
1    Mount   18
2  Abraham   22


In [3]:
dt1=pd.Series(my_data)
print(dt1)

Names    [peter, Mount, Abraham]
Age                 [20, 18, 22]
dtype: object


In [4]:
my_dic={
    'Monday':"20-07-2022",
    'Tuesday':"21-07-2022",
    'Wednesday':"22-07-2022",
    'Thursday':"23-07-2022",
    'Friday':"24-07-2022",
    'Satursday':"25-07-2022",
    'Sunday':"26-07-2022"}
dt2=pd.Series(my_dic)
print(dt2)

Monday       20-07-2022
Tuesday      21-07-2022
Wednesday    22-07-2022
Thursday     23-07-2022
Friday       24-07-2022
Satursday    25-07-2022
Sunday       26-07-2022
dtype: object


In [5]:
my_dic1={
    'monday':my_dic['Monday'],
    'tuesday':my_dic['Tuesday']}
my_dic2=pd.Series(my_dic1)
my_dic2

monday     20-07-2022
tuesday    21-07-2022
dtype: object

# Analyzing Data / Cleaning Data Exercise

In [6]:
#loading mine.csv file
df1=pd.read_csv('mine.csv')
df1

Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
..        ...    ...       ...       ...
164        60    105       140     290.8
165        60    110       145     300.0
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4

[169 rows x 4 columns]

# Using head()method to get first 6 rows

In [7]:
df1.head(6)

Duration  Pulse  Maxpulse  Calories
0        60    110       130     409.1
1        60    117       145     479.0
2        60    103       135     340.0
3        45    109       175     282.4
4        45    117       148     406.0
5        60    102       127     300.0

# Using tail()method to get last 6 rows

In [8]:
df1.tail(6)

Duration  Pulse  Maxpulse  Calories
163        45    100       140     280.9
164        60    105       140     290.8
165        60    110       145     300.0
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4

# info() method show the summary of the dataset

In [9]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 169 entries, 0 to 168
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Duration  169 non-null    int64  
 1   Pulse     169 non-null    int64  
 2   Maxpulse  169 non-null    int64  
 3   Calories  164 non-null    float64
dtypes: float64(1), int64(3)
memory usage: 5.4 KB


# isnull()is used to check for null values

In [10]:
df1.isnull().sum().sort_values(ascending=False)


Calories    5
Duration    0
Pulse       0
Maxpulse    0
dtype: int64

# dropna() is used to remove rows or columns with missing values or NaN

In [11]:
df1.dropna()

Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
..        ...    ...       ...       ...
164        60    105       140     290.8
165        60    110       145     300.0
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4

[164 rows x 4 columns]

# fillna() Fills all missing values with a specific value

In [12]:
df1=df1.fillna(0)
df1

Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
..        ...    ...       ...       ...
164        60    105       140     290.8
165        60    110       145     300.0
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4

[169 rows x 4 columns]

# Remove rows with Duration, Pulse, Maxpulse & Calories that are not an int. Reduce rows from 169 to 164

In [13]:
df1 = df1.loc[pd.to_numeric(df1['Duration'], errors='coerce').notnull()] 
df1 = df1.loc[pd.to_numeric(df1['Pulse'], errors='coerce').notnull()] 
df1 = df1.loc[pd.to_numeric(df1['Maxpulse'], errors='coerce').notnull()] 
df1 = df1.loc[pd.to_numeric(df1['Calories'], errors='coerce').notnull()]
df1

Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
..        ...    ...       ...       ...
164        60    105       140     290.8
165        60    110       145     300.0
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4

[169 rows x 4 columns]

# duplicated() Returns a boolean Series indicating duplicate rows

In [14]:

df1.duplicated()

0      False
1      False
2      False
3      False
4      False
       ...  
164    False
165    False
166    False
167    False
168    False
Length: 169, dtype: bool

# drop_duplicates() Removes duplicate rows

In [15]:

df1=df1.drop_duplicates()
df1

Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
..        ...    ...       ...       ...
164        60    105       140     290.8
165        60    110       145     300.0
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4

[162 rows x 4 columns]

# corr()  Calculates the correlation between columns using different such as kendall etc

In [16]:

df1.corr(method='kendall')

Duration     Pulse  Maxpulse  Calories
Duration  1.000000 -0.136995 -0.061297  0.656267
Pulse    -0.136995  1.000000  0.637570  0.149123
Maxpulse -0.061297  0.637570  1.000000  0.181291
Calories  0.656267  0.149123  0.181291  1.000000

In [17]:
#loading work.csv file
df2=pd.read_csv('Work.csv')
df2

Name         city    age py-score
0    Emma      Kampala     23       90
1  Wilber        Mbale     26       75
2   Robin         Gulu     25     mine
3   Tevor  Livingstone    NaN       89
4    Yeko       Tororo     20       94
5  Miriam         Arua  Train      NaN
6   Jesca      Mbarara     21       84

In [18]:
df2.head()

Name         city  age py-score
0    Emma      Kampala   23       90
1  Wilber        Mbale   26       75
2   Robin         Gulu   25     mine
3   Tevor  Livingstone  NaN       89
4    Yeko       Tororo   20       94

In [19]:
df2.tail()

Name         city    age py-score
2   Robin         Gulu     25     mine
3   Tevor  Livingstone    NaN       89
4    Yeko       Tororo     20       94
5  Miriam         Arua  Train      NaN
6   Jesca      Mbarara     21       84

In [20]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7 entries, 0 to 6
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   Name      7 non-null      object
 1   city      7 non-null      object
 2   age       6 non-null      object
 3   py-score  6 non-null      object
dtypes: object(4)
memory usage: 352.0+ bytes


In [21]:
df2.dropna()

Name     city age py-score
0    Emma  Kampala  23       90
1  Wilber    Mbale  26       75
2   Robin     Gulu  25     mine
4    Yeko   Tororo  20       94
6   Jesca  Mbarara  21       84

In [22]:
df2=df2.fillna(0)


# removing wrong data in the aga column

In [23]:
df2 = df2[pd.to_numeric(df2['age'], errors='coerce').notnull()] 
df2

Name         city age py-score
0    Emma      Kampala  23       90
1  Wilber        Mbale  26       75
2   Robin         Gulu  25     mine
3   Tevor  Livingstone   0       89
4    Yeko       Tororo  20       94
6   Jesca      Mbarara  21       84

# removing wrong data in the py-score column

In [24]:
df2 = df2[pd.to_numeric(df2['py-score'], errors='coerce').notnull()] 
df2

Name         city age py-score
0    Emma      Kampala  23       90
1  Wilber        Mbale  26       75
3   Tevor  Livingstone   0       89
4    Yeko       Tororo  20       94
6   Jesca      Mbarara  21       84

In [25]:
df2.duplicated()

0    False
1    False
3    False
4    False
6    False
dtype: bool

In [26]:
df2.drop_duplicates()

Name         city age py-score
0    Emma      Kampala  23       90
1  Wilber        Mbale  26       75
3   Tevor  Livingstone   0       89
4    Yeko       Tororo  20       94
6   Jesca      Mbarara  21       84

In [27]:
df2.corr(method='kendall', numeric_only=True)

Empty DataFrame
Columns: []
Index: []